bolt-modules/boltlib/tasks/apply_catalog.rb

```diff
@@ -0,0 +1,56 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require 'json'
+require 'puppet'
+require 'puppet/configurer'
+
+Puppet.initialize_settings([])
+run_mode = Puppet::Util::RunMode[:user]
+Puppet.settings.initialize_app_defaults(Puppet::Settings.app_defaults_for_run_mode(run_mode))
+
+Puppet::ApplicationSupport.push_application_context(run_mode, :local)
+
+# Avoid extraneous output
+Puppet[:summarize] = false
+
+# Don't interfere with the normal agent
+Puppet[:report] = false
+Puppet[:statefile] = Tempfile.new('statefile')
```

Comment:
I don't think this worked for me, I had to create a Tempfile, then call `path` on it.